src/AppInstallerTestExeInstaller/main.cpp

```diff
@@ -14,7 +14,7 @@ using namespace std::filesystem;
 std::string_view registrySubkey = "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\";
 std::string_view defaultProductID = "{A499DD5E-8DC5-4AD2-911A-BCD0263295E9}";
 
-std::wstring GenerateUninstaller(std::ostream& out, const path& installDirectory) {
+std::wstring GenerateUninstaller(std::ostream& out, const path& installDirectory, const std::wstring& productID) {

```

Comment:
>std::wstring [](start = 0, length = 12)

Not your fault either, but it's suggested to return path here and WriteToUninstallRegistry() below(line 46) should take in std::filesystem::Path as input